dd-sdk-android-core/src/test/kotlin/com/datadog/android/core/DatadogCoreTest.kt

```diff
@@ -235,6 +235,36 @@ internal class DatadogCoreTest {
         )
     }
 
+    @Test
+    fun `M use immutable values W setUserInfo()`(
```

Comment:
> [!IMPORTANT]
> We don't have the bug right now but it might be relevant to also test that we don't allow mutation when calling `addUserProperties()`